src/code/client.js

```diff
@@ -408,11 +415,16 @@ class CloudFileManagerClient {
   }
 
   openSharedContent(id) {
+    const { shareProvider } = this.state
     this._event('willOpenFile', {op: "openSharedContent"})
-    return (this.state.shareProvider != null ? this.state.shareProvider.loadSharedContent(id, (err, content, metadata) => {
-      if (err) { return this.alert(err, () => this.ready()) }
-      return this._fileOpened(content, metadata, {overwritable: false, openedContent: content.clone()})
-  }) : undefined)
+    if(shareProvider.loadSharedContent) {
+      shareProvider.loadSharedContent(id, (err, content, metadata) => {
+        if (err) {
```

Comment:
Might be cleaner to use `if/else` and then remove the returns since this is inside a callback.